### PR TITLE
KTOR-4281 Use only protocol from `DefaultRequest` for requests with host set

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

File: ktor-client/ktor-client-core/common/test/DefaultRequestTest.kt [hunk 1]
```diff
@@ -111,14 +112,34 @@ class DefaultRequestTest {
         val defaultUrl = Url(URLBuilder.Companion.origin)
         if (defaultUrl.port == defaultUrl.protocol.defaultPort) {
             assertEquals("https://localhost", client.get {}.bodyAsText())
-            assertEquals("ws://other.host:443/", client.get("ws://other.host/").bodyAsText())
```
Comment (e5l) [inline ktor-client/ktor-client-core/common/test/DefaultRequestTest.kt:114]: could you tell me why do we drop these checks?

Comment (e5l) [inline ktor-client/ktor-client-core/common/test/DefaultRequestTest.kt:114]: could you update these checks to present values instead of removing them?

**Review by e5l — APPROVED** [review]
lgtm

File: ktor-client/ktor-client-core/common/test/DefaultRequestTest.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-client/ktor-client-core/common/test/DefaultRequestTest.kt:114]: It was not dropped, but moved outside the `if`.

**Pull request merged**